NetExercise/Basic/Exercise17.cs

```diff
@@ -0,0 +1,44 @@
+namespace NetExercise.Basic
+{
+    public class Exercise17
+    {
+        public static int FindNumber(int n)
+        {
+            var num = n;
+
+            while(!(Prime(num) && SymmetricNumber(num)))
+            {
+                num++;
+            }
+
+            return num;
+        }
+
+        public static bool Prime(int n)
+        {
+            if (n < 2) return false;
+            
+            for (int i = 2; i <= n/2; i++)
```

Comment:
Bình thường có người sẽ viết
```csharp
            for (int i = 2; i < n; i++)
```
Ví dụ n là 1 tỷ thì vòng lặp này chạy khoản 1 tỉ lần lần. 

Với cách optimize của anh thì còn khoảng 500 triệu lần

```csharp
            for (int i = 2; i <= n/2; i++)
```

Thật ra việc optimize đó về mặt tin học là không có ý nghĩa lắm vì tính theo tỷ lệ thì giảm được 50%.

Với lại, ở đây, có thể thay dấu `<= ` thành `<`, nó sẽ nhìn đẹp hơn (mặc dù không có tác dụng mấy).

Có một cách khác tốt hơn để code chạy nhanh hơn hẳn. Đó là thế này

Tưởng tượng n chia hết cho a thì n cũng chia hết cho (n/a). Ví dụ 8 chia hết cho 2 thì cũng sẽ chia hết cho (8/2=4)

Gọi x là căn bậc 2 của n. Nếu n chia hết cho 1 số nào đó từ x đến n-1 thì chắc chắn là n cũng phải chia hết 1 số nào đó từ 2 đến x.

Hay nói cách khác là nếu không tìm thấy số nào từ 2 đến x mà n chia hết thì cũng khỏi cần tìm từ x đến n-1 luôn vì không có đâu.

i < x tương đương với i*i < n, 

```suggestion
            for (int i = 2; i * i < n; i++)
```

Với n là 1 tỉ thì vòng for trên chỉ lặp khoảng 32 ngàn lần. Anh hiểu 32 ngàn so với 500 triệu thì nó chênh lệch nhau cỡ nào rồi đấy.